src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -538,7 +543,7 @@ public OpenIdConnectAuthenticationHandler(HttpClient backchannel)
             if (userInfoSubjectClaimValue == null || !string.Equals(userInfoSubjectClaimValue, subjectClaimType.Value, StringComparison.Ordinal))
             {
                 Logger.LogError(Resources.OIDCH_0039_Subject_Claim_Mismatch);
-                return ticket;
+                return null;
```

Comment:
Consider faulting here as this is a protocol error. Returning 'null' will give the next middleware a crack at the message.